File: C++/factorial.cpp
#include <iostream>
int main()
{
    int factorial_of, result = 1;
    std::cout << "Hey man! Give me a number and I'll return you the factorial.\n";
    std::cin >> factorial_of;
    for(int i = 1; i <= factorial_of; i++)
    {
        result *= i;
    }
    printf("The factorial of %d is = %d\n", factorial_of, result);
}

File: Ruby/factorial.rb
puts "Hey man! Give me a number and I'll return you the factorial.\n"
factorial_of = gets.chomp
begin
    factorial_of = Integer(factorial_of)
    result = 1
    for i in 1..factorial_of
        result *= i
    end
    print "The factorial of ", factorial_of, " is = ", result, "\n"
rescue
    puts "I asked you for a number..."
end
